src/ServiceContainer/ServiceProvider/ServiceContainerServiceProvider.php

```diff
@@ -230,6 +234,24 @@ public function alterContainerDefinition(&$container_definition) {
   }
 
   /**
+   * Automatically register all annotated Plugins.
+   *
+   * @param array $container_definition
+   *   The container definition to process.
+   * @param array $definition
+   *   The parameter definition.
+   */
+  public function registerAnnotatedPluginTypes(&$container_definition, $parameter_definitions) {
+    foreach($parameter_definitions as $definition) {
+      $owner = $definition['owner'];
+      $type = $definition['type'];
+
+      $container_definition['services'][$owner . '.' . $type] = array();
```

Comment:
We need the alias support here as well ... (can we put it from below CTools into a helper method?)